docs/events/glides/com.mbta.ctd.glides.trips_updated.v1.mdx

```diff
@@ -0,0 +1,288 @@
+
+Fields in the object:
+
+- `type` (string): `"updated"|"added"`. If this field is `"updated"`, the object is this TripUpdated object. If it's `"added"`, it's a [TripAdded](#tripadded) object that extends TripUpdated and has all these fields but with additional meaning, fields, and restrictions, see above. Subsequent updates to previously-added trips have `type` `"updated"`.
+- `tripKey` ([TripKey](#tripkey)): which trip is being updated.
+- `comment` (string, optional): free text information about the trip. Could potentially be the empty string, if a comment was deleted.
+- `startLocation` ([Location](.#location) | "unset", optional): the new destination of the train.
+- `endLocation` ([Location](.#location) | "unset", optional): the new destination of the train.
+- `startTime` ([Time](#time) | "unset", optional): if present, the new time that the train is expected to depart `startLocation` (or the existing `startLocation` of the trip).
+- `endTime` ([Time](#time) | "unset", optional): if present, the new time that the train is expected to arrive at `endLocation` (or the existing `endLocation` of the trip).
+- `cars` (array of [Car](#car), optional): array of length 1 or 2, containing the car numbers and operators for each car in the train assigned to the trip. If absent, there are no changes to any cars. If present, the length of the array is the length of the train. In a two car train, the front car is listed first.
+- `dropped` ([DroppedReason](#droppedreason) | false, optional): If present the trip was dropped for the provided reason. If set to `false`, the trip is not dropped (and restored if previously dropped).
+- `scheduled` ([Scheduled](#scheduled) | null): For trips in Glides' schedule, this represents the scheduled data for the trip. It will always be present and won't change between subsequent updates to the same trip. For added trips, this will always be `null`.
+
+Some values might be the special string `"unset"`. This means to treat the field as if it had never been modified. This is semantically slightly different than setting the value to the scheduled value. For example, setting `startTime` to `"unset"` implies we don't know when the trip will leave, and the scheduled time might be a good guess, but setting `startTime` to equal the scheduled start time implies that an inspector has confirmed the trip will leave at that time.
+
+_Notes:_
+
+- setting a new location or time does not modify the [TripKey](#tripkey) for a scheduled trip.
+- If `cars` is updated from a 1-element list to a 2-element list, then Glides SHOULD include data for the 2nd car. This is relevant if a two-car train has a car removed and then re-added. If a field was set before the car was dropped, then when the car is restored, clients MUST assume that the field is `"none"` as opposed to retaining its previous value, but Glides SHOULD include the data to avoid the ambiguity.
```

Comment:
suggestion: in this documentation, `SHOULD` is probably not useful for consumers, as they can't rely on it. Does Glides provide this data, or not?

Reply:
Good point. I've gone through a reworded a couple instances of "Glides SHOULD" to be more certain where I know what we'll do.

But in some cases, like this one, I don't want to promise too much and I don't want clients to rely on it. "SHOULD" is still the right way to express the contract that we'll try to follow this, but we might not, and we want to be able to change the behavior if we have a good reason without clients breaking.

For reference, [RFC keywords definition](https://www.rfc-editor.org/rfc/rfc2119) for SHOULD:
> This word, or the adjective "RECOMMENDED", mean that there may exist valid reasons in particular circumstances to ignore a particular item, but the full implications must be understood and carefully weighed before choosing a different course.

Examples where it would be wrong to remove the SHOULD and imply we're certain, so I kept it:

- In TripAdded: "The event SHOULD only include values which were explicitly included by the author: inferred values SHOULD NOT be included." We currently have [a bug](https://app.asana.com/0/616151179860796/1204649263600516/f) where we include some inferred data. The data is still usable, though, and "the bug means we're misleading about how reliable this data is" is a less disasterous outcome than "the bug means we've broken this contract".

- In TripAdded "`dropped` SHOULD NOT be set". It would be weird if was set, but I can't promise that it would never happen. Like what if during a disruption where trains are running from an unscheduled endpoint, someone is adding trips as they go. They're adding trips at a regular headway, but then end up short an operator and can't run the next trip. They add the trip to the trainsheet to mark where they would have sent it, and immediately drop and write in a reason. Semantically, that's a trip that's both added and dropped at the same time and it's fine.